pkg/controller/studyjob/studyjob_controller.go

```diff
@@ -117,7 +121,8 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 			OwnerType:    &katibv1alpha1.StudyJob{},
 		})
 	if err != nil {
```

Comment:
invalidCRDResources should be updated only when CRD is not installed. We should explicitly check whether CRD exists or not. If watch fails for any other reason(even when CRD is installed), controller should fail by raising error. 

You can use crd clientset  to check. 
 https://github.com/kubeflow/tf-operator/blob/master/cmd/tf-operator.v1beta1/app/server.go#L192

Reply:
OK, I will handle it